libtuf/src/main/scala/com/advancedtelematic/libtuf/reposerver/UserReposerverClient.scala

```diff
@@ -78,18 +84,19 @@ class UserReposerverHttpClient(reposerverUri: URI,
     val req = Http(apiUri("targets.json")).method("GET")
     execHttp[SignedPayload[TargetsRole]](req)().map {
       case HttpResponse(payload, response) =>
-        val etag = response.header("ETag").map(_.replace("W/", "")).map(ETag.apply)
-        TargetsResponse(payload, etag)
+        val checksumO = response.header("x-ats-role-checksum").flatMap { v => refineV[ValidChecksum](v).toOption }
+        TargetsResponse(payload, checksumO)
     }
   }
 
-  def pushTargets(role: SignedPayload[TargetsRole], etag: Option[ETag]): Future[Unit] = {
+  def pushTargets(role: SignedPayload[TargetsRole], previousChecksum: Option[Refined[String, ValidChecksum]]): Future[Unit] = {
     val put = Http(apiUri("targets")).method("PUT")
-    val req = etag.map(e => put.header("If-Match", e.value)).getOrElse(put)
+    val req = previousChecksum.map(e => put.header("x-ats-role-checksum", e.value)).getOrElse(put)
+
 
     execJsonHttp[Unit, SignedPayload[TargetsRole]](req, role) {
-      case 412 | 428 =>
-        Future.failed(EtagNotValid)
+      case 412 | 428 => // TODO: Check error codes instead ?
```

Comment:
Yeah would look better with symbolic error codes.

Reply:
At this point I don't have the response just the code. Maybe I can change that.

Reply:
Doesn't `akka.http.scaladsl.model.StatusCodes` contain the codes?

Reply:
akka is not a dependency of cli